src/reducer/taskReducer.ts

```diff
@@ -0,0 +1,29 @@
+import { TaskAction, todo, TaskActionType } from "../feature/Types";
+
+export default function tasksReducer(
+  tasks: todo[],
+  action: TaskAction
+): todo[] {
+  switch (action.type) {
+    case TaskActionType.ADD: {
+      return [
+        ...tasks,
+        {
+          completed: action.payload.completed,
+          content: action.payload.content,
+        },
+      ];
+    }
+    case TaskActionType.DELETE: {
+      return tasks.filter((task: todo, i: number) => action.index !== i);
```

Comment:
separate each case into its own function. this will make the code more modular and easier to read